src/download.rs

```diff
@@ -14,11 +16,23 @@ where
     U: reqwest::IntoUrl,
     W: io::Write,
 {
+    // All HTTP status codes that must result in an immediate failure.
+    let failure_status = [
+        StatusCode::FORBIDDEN, // 403
+        StatusCode::NOT_FOUND, // 404
+    ];
+
     #[rustfmt::skip]
     let mut res = client.get(url)
         .send()
         .await?;
 
+    // Return immediately on download failure on the client side.
+    let status = res.status();
+    if status.is_client_error() && failure_status.contains(&status) {
```

Comment:
probably better to just do a match statement for the `failure_status` checking – there's no real benefit to having the statuses in an array and also `contains()` is going to do a linear search through the array whereas a match statement wouldn't

Reply:
Ok, updated as you suggested.